storm-server/src/main/java/org/apache/storm/localizer/Localizer.java

```diff
@@ -226,9 +218,8 @@ public boolean accept(File dir, String name) {
   protected void reconstructLocalizedResources() {
     try {
       LOG.info("Reconstruct localized resource: " + getUserCacheDir().getPath());
-      File[] users = readDirContents(getUserCacheDir().getPath());
-
-      if (users != null) {
+      Collection<File> users = ConfigUtils.readDirFiles(getUserCacheDir().getPath());
+      if (users == null || users.isEmpty()) {
```

Comment:
This can be reverted.  The only need was to check for null so we don't NPE.  This inverted the check.